plugins/modules/foreman_template_sync.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Anton Nesterov (@nesanton)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_template_sync
+short_description: Sync templates from or to a repository
+description:
+  - Sync provisioning templates, partition tables and job templates from external git repository and/or file system.
+  - Based on foreman_templates plugin U(https://github.com/theforeman/foreman_templates).
+  - Some defaults can be set in TemplateSync settings using M(foreman_setting) or GUI.
+  - Module attempts to be idempotent as much as the plugin allows.
+  - "Known issues:"
+  - I(direction=export) exports all templates even if only one changes, there also could be 500 responses on C(export).
+author:
+  - "Anton Nesterov (@nesanton)"
+options:
+  direction:
+    description:
+      - Direction of the sync.
```

Comment:
I wonder if this should become `state: imported/exported` instead of a new param

Reply:
Maybe it's better to call it an action module (a bit like katello_sync) that has no state at all. This kind of falls in line with the fact that there is no proper check_mode.

Reply:
is there a special way to make it so? I seem to be inheriting `state` in argument spec

Reply:
I should be possible if you inherit from `ForemanAnsibleModule`.

Reply:
ah right